src/main/java/org/elasticsearch/cluster/service/InternalClusterService.java

```diff
@@ -332,6 +332,10 @@ public int numberOfPendingTasks() {
         return updateTasksExecutor.getNumberOfPendingTasks();
     }
 
+    @Override
+    public TimeValue getPendingTaskTimeInQueue() {
+        return updateTasksExecutor.getPendingTaskTimeInQueue();
```

Comment:
I think this should be implemented by scanning all tasks in the queue since the first one isn't really the max value which is what we care about. Also it would be great if it can share the task normalization code in `pendingTasks`